cmd/dockerd/daemon.go

```diff
@@ -423,6 +428,7 @@ func initBuildkit(ctx context.Context, d *daemon.Daemon) (_ builderOptions, clos
 			Exported: d.ImageExportedByBuildkit,
 			Named:    d.ImageNamedByBuildkit,
 		},
+		CDISpecDirs: cdiSpecDirs,
```

Comment:
Does  `CDISpecDirs=nil` disable CDI, or does it pick the defaults from BuildKit?

Reply:
CDI is enabled by default in BuildKit and if specs dir is empty it would fail similar to https://github.com/moby/buildkit/pull/5726. Does moby disabled CDI if specs dir is empty?

Reply:
it disables CDI by default, but if it's enabled, it would pick CDI's default directories. So if there's a similar default on BuildKit's side, it would mean that _not_ specifying directories would potentially pick different  defaults for BuildKit than for Docker Engine https://github.com/moby/moby/blob/b570831cc3a3fcfe4edc96af4c249199b019c7dd/cmd/dockerd/daemon.go#L634-L644

> CDI is enabled by default in BuildKit 

Is there an option to disable it? It feels like it's unexpected for the docker daemon to not enable the feature, but BuildKit to enable it anyway.

Reply:
> but if it's enabled, it would pick CDI's default directories

Ok then I think we should pick `cdi.DefaultSpecDirs...` as well on Moby.

> it disables CDI by default

I wonder if CDI could be enabled by default on Moby? Has this question been put on the table?

Reply:
BuildKit uses docker config (features+dirs) when running as part of dockerd.

This means that atm it is opt-in in `dockerd` and opt-out in `buildkitd`. We should probably stop it from being opt-in in general in Moby unless there are some issues I don't know about, but that is better in follow-up.

Reply:
Yeah, I think opt-out can be discussed; I think it's still useful to have a control to opt-out, but we can enable by default (in a follow-up)